.github/workflows/contracts.yml

```diff
@@ -48,14 +60,89 @@ jobs:
       - name: Deploy contracts
         run: yarn deploy
 
+  contracts-deployment-testnet:
+    needs: [contracts-build-and-test]
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Load environment variables
+        uses: keep-network/load-env-variables@v1
+        with:
+          environment: ${{ github.event.inputs.environment }}
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Get upstream packages' versions
+        uses: keep-network/upstream-builds-query@v1
+        id: upstream-builds-query
+        with:
+          upstream-builds: ${{ github.event.inputs.upstream_builds }}
+          query: |
+            keep-core-solidity-version = github.com/keep-network/keep-core/solidity#version
+            tbtc-solidity-version = github.com/keep-network/tbtc/solidity#version
+
+      - name: Resolve latest contracts
+        run: |
+          yarn add \
+            @keep-network/keep-core@${{ steps.upstream-builds-query.outputs.keep-core-solidity-version }} \
+            @keep-network/tbtc@${{ steps.upstream-builds-query.outputs.tbtc-solidity-version }}
+
+      - name: Deploy contracts
+        run: yarn deploy --network ${{ github.event.inputs.environment }}
```

Comment:
We need to make sure two env variables are set: `CHAIN_API_URL` and `CONTRACT_OWNER_ACCOUNT_PRIVATE_KEY`.

https://github.com/keep-network/coverage-pools/blob/2c290477c34d460dfb2186cfd7bf1f0d6b517672/hardhat.config.ts#L42-L44

Reply:
I've added passing of env variables in the deploy step (see b4d31fafeeb7c9597e490b33a1ced6cc811a2ad1) , but something is still off.  I'm still getting the `failed to get chainId, falling back on net_version...` error (see https://github.com/keep-network/coverage-pools/runs/3172830878?check_suite_focus=true).
I'll try to find the root cause.